### ISSUE TITLE
expand lines if less than max width

### ISSUE THREAD

**Issue by AlexisQapa** [nicklockwood/SwiftFormat#553]
Hi,

Could it be possible to expand lines that are less than maxwidth ?

Let's say i have a func:
I add another param, it goes over maxwidth
-> swift format compact it
I remove the param 
-> go back to inline mode

I hope my message clear, not sure thought

**Comment by nicklockwood** [comment]
@AlexisQapa this is something I plan to add once the line shortening is working well. It will be a separate rule so it can be disabled if people don't want it.

**Comment by AlexisQapa** [comment]
Awesome ! 

I there non public work being done or ideas on this or the shortening rules ? If I can find some time I'd like to contribute

**Comment by nicklockwood** [comment]
@AlexisQapa nothing non-public. If you have ideas then you can post them here, or open a PR and we can iterate on them.

I did nothing on this for years until I came to the conclusion that it's better to ship a partial solution and then iterate on it. As long as it doesn't break anyone's project and/or is easy to turn off then I'd  be happy to accept a PR that just handles one or two cases and build on that.

**Comment by rgoldberg** [comment]
@calda Was this actually completed?

**Comment by nicklockwood** [comment]
I don't think it was - it might have been incorrectly flagged due to the wrap-threshold feature maybe?

**Comment by nicklockwood** [comment]
Actually that wouldn't explain why it was marked as fixed in develop on March 10

**Comment by calda** [comment]
There was a bug in the "fixed in develop" automation. The collapsed release notes in https://github.com/nicklockwood/SwiftFormat/pull/2453 incorrectly triggered issues in this repo to be marked as fixed ([logs here](https://github.com/nicklockwood/SwiftFormat/actions/runs/22884711675/job/66394769717)). Landed some improvements to the automation in https://github.com/nicklockwood/SwiftFormat/pull/2515.

**Comment by rgoldberg** [comment]
Thanks for looking into this.
